### ISSUE TITLE
Unable to Download

### ISSUE THREAD

**Issue by Ayush-Singh2310** [Zackptg5/Cross-Compiled-Binaries-Android#43]
Module fails at "checking internet connection phase"
Log:

- Copying zip to temp directory
- Installing Cross_Compiled_Binaries-v8.0(10).zip
- Device is system-as-root
Archive:  /data/user/0/cu.mu.fnr.k.o.t/cache/flash/install.zip
  inflating: module.prop
*************************
 Cross Compiled Binaries 
 by Zackptg5 
*************************
*******************
 Powered by Magisk 
*******************
Archive:  /data/user/0/cu.mu.fnr.k.o.t/cache/flash/install.zip
  inflating: customize.sh
- Extracting module files
Archive:  /data/user/0/cu.mu.fnr.k.o.t/cache/flash/install.zip
  inflating: .gitattributes
  inflating: .gitignore
  inflating: README.md
  inflating: busybox-arm
  inflating: busybox-x86
  inflating: curl-arm
  inflating: curl-x86
  inflating: customize.sh
  inflating: functions.sh
  inflating: module.prop
  inflating: wg-arm
  inflating: wg-quick-arm
  inflating: wg-quick-x86
  inflating: wg-x86
+ true
+ '[' -z arm ]
+ set_perm /data/adb/modules_update/ccbins/busybox-arm 0 0 0755
+ chown 0:0 /data/adb/modules_update/ccbins/busybox-arm
+ chmod 0755 /data/adb/modules_update/ccbins/busybox-arm
+ CON=
+ '[' -z ]
+ CON=u:object_r:system_file:s0
+ chcon u:object_r:system_file:s0 /data/adb/modules_update/ccbins/busybox-arm
+ alias 'awk=/data/adb/modules_update/ccbins/busybox-arm awk'
+ alias 'grep=/data/adb/modules_update/ccbins/busybox-arm grep'
+ alias 'md5sum=/data/adb/modules_update/ccbins/busybox-arm md5sum'
+ alias 'ping=/data/adb/modules_update/ccbins/busybox-arm ping'
+ mv -f /data/adb/modules_update/ccbins/curl-arm /data/adb/modules_update/ccbins/curl
+ set_perm /data/adb/modules_update/ccbins/curl 0 0 0755
+ chown 0:0 /data/adb/modules_update/ccbins/curl
+ chmod 0755 /data/adb/modules_update/ccbins/curl
+ CON=
+ '[' -z ]
+ CON=u:object_r:system_file:s0
+ chcon u:object_r:system_file:s0 /data/adb/modules_update/ccbins/curl
+ mv -f /data/adb/modules_update/ccbins/wg-arm /data/adb/modules_update/ccbins/wg
+ set_perm /data/adb/modules_update/ccbins/wg 0 0 0755
+ chown 0:0 /data/adb/modules_update/ccbins/wg
+ chmod 0755 /data/adb/modules_update/ccbins/wg
+ CON=
+ '[' -z ]
+ CON=u:object_r:system_file:s0
+ chcon u:object_r:system_file:s0 /data/adb/modules_update/ccbins/wg
+ mv -f /data/adb/modules_update/ccbins/wg-quick-arm /data/adb/modules_update/ccbins/wg-quick
+ set_perm /data/adb/modules_update/ccbins/wg-quick 0 0 0755
+ chown 0:0 /data/adb/modules_update/ccbins/wg-quick
+ chmod 0755 /data/adb/modules_update/ccbins/wg-quick
+ CON=
+ '[' -z ]
+ CON=u:object_r:system_file:s0
+ chcon u:object_r:system_file:s0 /data/adb/modules_update/ccbins/wg-quick
+ '[' -f /data/adb/modules/ccbins/doh ]
+ alias 'wg=/data/adb/modules_update/ccbins/wg'
+ alias 'wg-quick=/data/adb/modules_update/ccbins/wg-quick'
+ '[' 30 -lt 23 ]
+ curlalias='/data/adb/modules_update/ccbins/curl -Ls'
+ '[' -f /data/adb/modules_update/ccbins/doh ]
+ pm list packages com.wireguard.android
+ '['  ]
+ . /data/adb/modules_update/ccbins/functions.sh
+ unset curl
+ alias 'curl=/data/adb/modules_update/ccbins/curl -Ls'
+ test_connection
+ ui_print '- Testing internet connection'
+ true
+ echo '- Testing internet connection'
+ /data/adb/modules_update/ccbins/curl -Ls --connect-timeout 3 -I https://www.google.com
+ /data/adb/modules_update/ccbins/busybox-arm grep -q 'HTTP/.* 200'
- Testing internet connection
+ /data/adb/modules_update/ccbins/busybox-arm ping -q -c 1 -W 1 google.com
+ return 0
+ '[' -f /data/adb/modules/ccbins/system/bin/ccbins ]
+ branch=master
+ download_file /data/adb/modules_update/ccbins/.checksums https://raw.githubusercontent.com/Zackptg5/Cross-Compiled-Binaries-Android/master/ccbins_files/checksums.txt
+ rm -f /data/adb/modules_update/ccbins/dlerror
+ local 'file=/data/adb/modules_update/ccbins/.checksums' 'url=https://raw.githubusercontent.com/Zackptg5/Cross-Compiled-Binaries-Android/master/ccbins_files/checksums.txt'
+ rm -f /data/adb/modules_update/ccbins/.checksums
+ /data/adb/modules_update/ccbins/curl -Ls -o /data/adb/modules_update/ccbins/.checksums https://raw.githubusercontent.com/Zackptg5/Cross-Compiled-Binaries-Android/master/ccbins_files/checksums.txt
+ '[' /data/adb/modules_update/ccbins/.checksums '==' /data/adb/modules_update/ccbins/.checksums ]
+ head -n1 /data/adb/modules_update/ccbins/.checksums
+ '['  '==' checksums.txt ]
+ return 1
+ '[' -z  ]
+ echo -e 'Alibaba\nhttps://dns.alidns.com/dns-query'
+ tail -n1 /data/adb/modules_update/ccbins/doh
+ curlalias='/data/adb/modules_update/ccbins/curl -Ls --doh-url https://dns.alidns.com/dns-query'
+ '[' -d /data/adb/modules/ccbins ]
+ . /data/adb/modules_update/ccbins/functions.sh
+ unset curl
+ alias 'curl=/data/adb/modules_update/ccbins/curl -Ls --doh-url https://dns.alidns.com/dns-query'
+ download_file /data/adb/modules_update/ccbins/.checksums https://raw.githubusercontent.com/Zackptg5/Cross-Compiled-Binaries-Android/master/ccbins_files/checksums.txt
+ rm -f /data/adb/modules_update/ccbins/dlerror
+ local 'file=/data/adb/modules_update/ccbins/.checksums' 'url=https://raw.githubusercontent.com/Zackptg5/Cross-Compiled-Binaries-Android/master/ccbins_files/checksums.txt'
+ rm -f /data/adb/modules_update/ccbins/.checksums
+ /data/adb/modules_update/ccbins/curl -Ls --doh-url https://dns.alidns.com/dns-query -o /data/adb/modules_update/ccbins/.checksums https://raw.githubusercontent.com/Zackptg5/Cross-Compiled-Binaries-Android/master/ccbins_files/checksums.txt
+ '[' /data/adb/modules_update/ccbins/.checksums '==' /data/adb/modules_update/ccbins/.checksums ]
+ head -n1 /data/adb/modules_update/ccbins/.checksums
+ '['  '==' checksums.txt ]
+ return 1
+ abort 'Unable to download files!'
+ ui_print 'Unable to download files!'
+ true
+ echo 'Unable to download files!'
+ true
+ '[' '!' -z /data/adb/modules_update/ccbins ]
+ rm -rf /data/adb/modules_update/ccbins
Unable to download files!
+ rm -rf /dev/tmp
+ exit 1
! Installation failed

**Comment by Zackptg5** [comment]
Referring to https://github.com/Magisk-Modules-Repo/ccbins/issues/6

**Comment by Ayush-Singh2310** [comment]
> Try it again. Either github had a hiccup or it's being blocked by something on your end such as a firewall

I tried it again and again and finally figured out that

Anything i download from github takes atleast 20~30 seconds to start

So is there a way i can increase the timeout time for the module??

**Comment by Ayush-Singh2310** [comment]
> Try it again. Either github had a hiccup or it's being blocked by something on your end such as a firewall

I tried it again and again and finally figured out that

Anything i download from github takes atleast 20~30 seconds to start

So is there a way i can increase the timeout time for the module??

**Comment by Zackptg5** [comment]
That's definitely not normal. You try using different dns service?
For the zip, change the curl-timeout number in functions.sh to whatever. Once installed, do the same in ccbins file in the module

**Comment by Ayush-Singh2310** [comment]
I found the reason. For some reason ccbin doesn't install if riru module is installed.
So i installed ccbins first and it worked.

**Comment by Ayush-Singh2310** [comment]
Found new issue, i haven't figured out which but 1 or more binaries cause the rom to be stuck in bootloop.
Currently using RN8P and AncientOS Android 12.

**Comment by Zackptg5** [comment]
That's unlikely an issue with the binary itself. Install then to a different location like /system/xbin so they don't overwrite any system ones. Idk which one it would be though, coreutils doesn't overwrite existing applets which is the only one I'd think would cause a bootloop

**Comment by Ayush-Singh2310** [comment]
> coreutils doesn't overwrite existing applets which is the only one I'd think would cause a bootloop
Already tried coreutils single install, worked without issues

**Comment by Ayush-Singh2310** [comment]
> Try it again. Either github had a hiccup or it's being blocked by something on your end such as a firewall

Solved this issue, it was simply an issue of selinux not being permissive.
I set it to permissive and it installed without issues.
Wish i knew this before trying out dozens of different custom rom's.
